integrations/tests/test_shuffle.py

```diff
@@ -0,0 +1,307 @@
+# Copyright (C) 2015, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute
+# it and/or modify it under the terms of GPLv2
+
+"""Unit tests for shuffle.py integration."""
+
+import sys
+import os
+import json
+import pytest
+import requests
+import shuffle
+from unittest.mock import patch, mock_open
+
+sys.path.append(os.path.join(os.path.dirname(
+    os.path.realpath(__file__)), '..', '..'))
+
+alert_template = {'timestamp': 'year-month-dayThours:minuts:seconds+0000',
+                  'rule': {'level': 0, 'description': 'alert description',
+                           'id': '',
+                           'firedtimes': 1},
+                  'id': 'alert_id',
+                  'full_log': 'full log.', 'decoder': {'name': 'decoder-name'},
+                  'location': 'wazuh-X'}
+
+msg_template = '{"severity": 1, "pretext": "WAZUH Alert", "title": "alert description", "text": "full log.", "rule_id": "rule-id", "timestamp": "year-month-dayThours:minuts:seconds+0000", "id": "alert_id", "all_fields": {"timestamp": "year-month-dayThours:minuts:seconds+0000", "rule": {"level": 0, "description": "alert description", "id": "rule-id", "firedtimes": 1}, "id": "alert_id", "full_log": "full log.", "decoder": {"name": "decoder-name"}, "location": "wazuh-X"}}'
+
+sys_args_template = ['/var/ossec/integrations/shuffle.py', '/tmp/shuffle-XXXXXX-XXXXXXX.alert', '',
+                     'http://<IP>:3001/api/v1/hooks/<HOOK_ID>', ' > /dev/null 2>&1']
+
+
+@pytest.mark.parametrize('args', [sys_args_template])
+def test_main_bad_arguments_exits(args):
+    """
+    Test that main function exits when wrong number of arguments are passed
+
+    Parameters
+    ----------
+    args: list[str]
+       list of the arguments passed to the main function
+
+    """
+    with patch("builtins.open", mock_open()), \
+            pytest.raises(SystemExit) as pytest_wrapped_e:
+        shuffle.main(args[0:2])
+    assert pytest_wrapped_e.type == SystemExit
```

Comment:
There is no need to perform this check . It is guaranteed that the exception will be a `SystemExit` one otherwise `pytest.raises` will fail and this assert sentence will not be executed.